internal/server/iigointernal/orchestration.go

```diff
@@ -0,0 +1,144 @@
+
+// iigoClients holds pointers to all the clients
+var iigoClients map[shared.ClientID]baseclient.Client
+
+// RunIIGO runs all iigo function in sequence
+func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.Client) error {
+
+	// TODO: Get Client pointers from gamestate https://imgur.com/a/HjVZIkh
+	iigoClients = *clientMap
+
+	// Initialise IDs
+	featureJudge.id = JudgeIDGlobal
+	featureSpeaker.id = SpeakerIDGlobal
+	featurePresident.id = PresidentIDGlobal
+
+	// Initialise iigointernal with their clientVersions
+	featureJudge.clientJudge = judgePointer
+	featurePresident.clientPresident = presidentPointer
+	featureSpeaker.clientSpeaker = speakerPointer
+
+	// Withdraw the salaries
+	errWithdrawPresident := featureJudge.withdrawPresidentSalary(g)
+	errWithdrawJudge := featureSpeaker.withdrawJudgeSalary(g)
+	errWithdrawSpeaker := featurePresident.withdrawSpeakerSalary(g)
+
+	// Handle the lack of resources
+	if errWithdrawPresident != nil {
+		return errors.Errorf("Could not run IIGO since President has no resoruces to spend")
+	}
+	if errWithdrawJudge != nil {
+		return errors.Errorf("Could not run IIGO since Judge has no resoruces to spend")
+	}
+	if errWithdrawSpeaker != nil {
+		return errors.Errorf("Could not run IIGO since Speaker has no resoruces to spend")
+	}
+
+	// Pay salaries into budgets
+	featureJudge.sendPresidentSalary()
+	featureSpeaker.sendJudgeSalary()
+	featurePresident.sendSpeakerSalary()
+
+	// 1 Judge actions - inspect history
+	_, judgeInspectingHistoryError := featureJudge.inspectHistory()
+
+	// 2 President actions
+	resourceReports := map[int]int{}
+	for _, v := range rules.VariableMap["islands_alive"].Values {
+		resourceReports[int(v)] = iigoClients[shared.ClientID(int(v))].ResourceReport()
+	}
+	featurePresident.broadcastTaxation(resourceReports)
+	featurePresident.requestAllocationRequest()
+	featurePresident.replyAllocationRequest(g.CommonPool)
+	featurePresident.requestRuleProposal()
+	ruleToVote := featurePresident.getRuleForSpeaker()
+
+	// 3 Speaker actions
+	featureSpeaker.SetRuleToVote(ruleToVote)
+	featureSpeaker.setVotingResult()
+	featureSpeaker.announceVotingResult()
+
+	// 4 Declare performance (Judge) (in future all the iigointernal)
+	if judgeInspectingHistoryError != nil {
+		featureJudge.declarePresidentPerformanceWrapped()
+
+		featureJudge.declareSpeakerPerformanceWrapped()
+	}
+
+	// Get new Judge ID
+	JudgeIDGlobal = featureSpeaker.appointNextJudge()
+	// Get new Speaker ID
+	SpeakerIDGlobal = featurePresident.appointNextSpeaker()
+	// Get new President ID
+	PresidentIDGlobal = featureJudge.appointNextPresident()
+
+	// Set judgePointer
+	judgePointer = iigoClients[shared.ClientID(JudgeIDGlobal)].GetClientJudgePointer()
+	// Set speakerPointer
+	speakerPointer = iigoClients[shared.ClientID(SpeakerIDGlobal)].GetClientSpeakerPointer()
+	// Set presidentPointer
+	presidentPointer = iigoClients[shared.ClientID(PresidentIDGlobal)].GetClientPresidentPointer()
```

Comment:
don't cast it like this, it's unsafe. Instead, return `shared.ClientID` from `appoint...`